webviz_subsurface/_datainput/well_completions.py

```diff
@@ -1,21 +1,30 @@
-from typing import Optional, Dict
+from typing import Optional, Dict, List, Tuple
 import json
+import re
 from pathlib import Path
 import glob
 
-import ecl2df
+from ecl2df import common
 
 
 def read_zone_layer_mapping(
     ensemble_path: str, zone_layer_mapping_file: str
-) -> Optional[Dict[int, str]]:
+) -> Tuple[Optional[Dict[int, str]], Optional[Dict[str, str]]]:
     """Searches for a zone layer mapping file (lyr format) on the scratch disk. \
     If one file is found it is parsed using functionality from the ecl2df \
     library.
     """
+    # pylint: disable=no-member
     for filename in glob.glob(f"{ensemble_path}/{zone_layer_mapping_file}"):
-        return ecl2df.EclFiles("").get_zonemap(filename=filename)
-    return None
+        zonelist = common.parse_lyrfile(filename=filename)
+        layer_zone_mapping = common.convert_lyrlist_to_zonemap(zonelist)
+        zone_color_mapping = {
+            zonedict["name"]: zonedict["color"]
+            for zonedict in zonelist
+            if "color" in zonedict and re.match("^#([A-Fa-f0-9]{6})", zonedict["color"])
```

Comment:
Do we want to (silently) filter out colors here? Add a `logging.WARNING`?

Reply:
yes, good idea. I needed to use logging.warning with lower case it seems.